pallets/gear/src/lib.rs

```diff
@@ -1316,6 +1316,7 @@ pub mod pallet {
         pub fn upload_code(origin: OriginFor<T>, code: Vec<u8>) -> DispatchResultWithPostInfo {
             let who = ensure_signed(origin)?;
 
+            // MIKITA: we can refund in case of hash already exists
```

Comment:
in case of existence we exit the execution due to `?` and error occurred there, no ways for refund

Reply:
got it